operators/test/e2e/test/apmserver/checks_apm.go

```diff
@@ -101,7 +127,96 @@ func (c *apmClusterChecks) CheckEventsAPI() test.Step {
 				assert.Equal(t, eventsErrorResponse.Accepted, 2)
 				assert.Len(t, eventsErrorResponse.Errors, 0)
 			}
-			// TODO: verify that the events eventually show up in Elasticsearch
 		},
 	}
 }
+
+// CountResult maps the result of a /index/_count request.
+type CountResult struct {
+	Count  int `json:"count"`
+	Shards struct {
+		Total      int `json:"total"`
+		Successful int `json:"successful"`
+		Skipped    int `json:"skipped"`
+		Failed     int `json:"failed"`
+	} `json:"_shards"`
+}
+
+// CheckEventsInElasticsearch checks that the events sent in the previous step have been stored.
+// We only count document to not rely on the internal schema of the APM Server.
+func (c *apmClusterChecks) CheckEventsInElasticsearch(apm apmtype.ApmServer, k *test.K8sClient) test.Step {
+	return test.Step{
+		Name: "Events should eventually show up in Elasticsearch",
+		Test: test.Eventually(func() error {
+			// Fetch the last version of the APM Server
+			var updatedApmServer apmtype.ApmServer
+			if err := k.Client.Get(k8s.ExtractNamespacedName(&apm), &updatedApmServer); err != nil {
+				return err
+			}
+
+			// Check that the metric has been stored
+			err := assertCountIndexEqual(
+				c.esClient,
+				fmt.Sprintf("apm-%s-metric-2017.05.30", updatedApmServer.Spec.Version),
+				1,
+			)
+			if err != nil {
+				return err
+			}
+
+			// Check that the error has been stored
+			err = assertCountIndexEqual(
+				c.esClient,
+				fmt.Sprintf("apm-%s-error-2018.08.09", updatedApmServer.Spec.Version),
+				1,
+			)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		}),
+	}
+}
+
+// assertCountIndexEqual asserts that the number of document in an index is the expected one, it raises an error otherwise.
+func assertCountIndexEqual(esClient client.Client, index string, expected int) error {
+	metricCount, err := countIndex(esClient, index)
+	if err != nil {
+		return err
+	}
+	if metricCount != expected {
+		return fmt.Errorf("%d document expected in index %s, got %d instead", expected, index, metricCount)
+	}
+	return nil
+}
+
+// countIndex counts the number of document in an index.
+func countIndex(esClient client.Client, indexName string) (int, error) {
+	r, err := http.NewRequest(
+		http.MethodGet, fmt.Sprintf("/%s/_count", indexName),
+		nil,
+	)
+	if err != nil {
+		return 0, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), DefaultReqTimeout)
+	defer cancel()
+	response, err := esClient.Request(ctx, r)
+	defer response.Body.Close() // nolint
```

Comment:
Should the `defer` to close the body be declared just after checking the error otherwise the response might be nil?